app/beacon-omop-worker/beacon_omop_worker/query_solvers.py

```diff
@@ -0,0 +1,255 @@
+from beacon_omop_worker.entities import (
+    ConditionOccurrence,
+    Person,
+    DrugExposure,
+    Measurement,
+    Observation,
+    ProcedureOccurrence,
+    Concept,
+    Vocabulary,
+)
+import beacon_omop_worker.config as config
+import logging
+import pandas as pd
+from sqlalchemy import (
+    select,
+)
+
+
+class FilterQuerySolver:
+    subqueries = list()
+
+    concept_table_map = {
+        "Condition": ConditionOccurrence,
+        "Ethnicity": Person,
+        "Drug": DrugExposure,
+        "Gender": Person,
+        "Race": Person,
+        "Measurement": Measurement,
+        "Observation": Observation,
+        "Procedure": ProcedureOccurrence,
+    }
+    concept_time_column_map = {
+        "Condition": ConditionOccurrence.condition_start_date,
+        "Ethnicity": Person.birth_datetime,
+        "Drug": DrugExposure.drug_exposure_start_date,
+        "Gender": Person.birth_datetime,
+        "Race": Person.birth_datetime,
+        "Measurement": Measurement.measurement_date,
+        "Observation": Observation.observation_date,
+        "Procedure": ProcedureOccurrence.procedure_date,
+    }
+    numeric_rule_map = {
+        "Measurement": Measurement.value_as_number,
+        "Observation": Observation.value_as_number,
+    }
+    boolean_rule_map = {
+        "Condition": ConditionOccurrence.condition_concept_id,
+        "Ethnicity": Person.ethnicity_concept_id,
+        "Drug": DrugExposure.drug_concept_id,
+        "Gender": Person.gender_concept_id,
+        "Race": Person.race_concept_id,
+        "Measurement": Measurement.measurement_concept_id,
+        "Observation": Observation.observation_concept_id,
+        "Procedure": ProcedureOccurrence.procedure_concept_id,
+    }
+
+    def __init__(self, db_manager: SyncDBManager) -> None:
+        self.db_manager = db_manager
+
+    def _get_concepts(self) -> pd.DataFrame:
+        """
+        Select vocabulary_id, concept_name, concept_id columns from Concept table
+        Returns:
+        concepts_df (pd.DataFrame): Concept table with relevant columns as a pandas dataframe.
+        """
+        concept_query = select(
+            Concept.vocabulary_id,
+            Concept.concept_id,
+            Concept.concept_name,
+        ).distinct()
+        concepts_df = pd.read_sql_query(
+            concept_query,
+            con=self.db_manager.engine.connect().execution_options(stream_results=True),
+        )
+        return concepts_df
+
+    def _get_table_concepts(self, query) -> pd.DataFrame:
+        """
+        Given a SQL query execute it and return the results in a pandas dataframe.
+        Args:
+            query: A SQL query to execute
+
+        Returns:
+            table_concepts_df (pd.DataFrame): A pandas dataframe containing the results of the query.
+        """
+        table_concepts_df = pd.read_sql_query(
+            query, con=self.db_manager.engine.connect()
+        )
+        return table_concepts_df
+
+    @staticmethod
```

Comment:
is there a reason this method is static? I'm assuming it's only used privately in the class if you've prefixed it with and underscore